text/0436-gamelift-l2.md

```diff
@@ -0,0 +1,945 @@
+  build: build
+});
+
+const queue = new gamelift.GameSessionQueue(this, 'Game session queue');
+queue.addDestination(fleet);
+```
+
+or
+
+```ts fixture=with-build
+import * as gamelift from 'aws-cdk-lib/aws-gamelift';
+
+const fleet = new gamelift.Fleet(this, 'Game server fleet', {
+  build: build
+});
+
+const alias = fleet.addAlias('live')
+
+const queue = new gamelift.GameSessionQueue(this, 'Game session queue');
+queue.addDestination(alias);
+```
+
+See [Setting up GameLift queues for game session placement](https://docs.aws.amazon.com/gamelift/latest/developerguide/realtime-script-uploading.html)
+in the *Amazon GameLift Developer Guide*.
+
+###### Setting notifications
+
+If you're using queues to manage game session placement in your game, you need a way to monitor the status of individual placement requests and take action as appropriate. Implementing event notifications is a fast and efficient method for tracking placement activity. If your game is in production, or in pre-production with high-volume placement activity, you should be using event notifications.
+
+There are two options for setting up event notifications. You can set up an SNS topic and have GameLift publish event notifications on placement activity by referencing the topic ID in a game session queue. Alternatively, you can use Amazon CloudWatch Events, which has a suite of tools available for managing events and taking action on them.
+
+
+```ts fixture=with-build
+import * as gamelift from 'aws-cdk-lib/aws-gamelift';
+
+const fleet = new gamelift.Fleet(this, 'Game server fleet', {
+  build: build
+});
+
+const topic = new sns.Topic(this, 'Topic');
+
+const queue = new gamelift.GameSessionQueue(this, 'Game session queue', {
+  notification: topic
+});
+queue.addDestination(fleet);
+```
+
+##### Managing game servers launch configuration
+
+GameLift uses a fleet's runtime configuration to determine the type and number of processes to run on each instance in the fleet. At a minimum, a runtime configuration contains one server process configuration that represents one game server executable. You can also define additional server process configurations to run other types of processes related to your game. Each server process configuration contains the following information:
+
+* The file name and path of an executable in your game build.
+
+* Optionally Parameters to pass to the process on launch.
+
+* The number of processes to run concurrently.
+
+A GameLift instance is limited to 50 processes running concurrently.
+
+```ts fixture=with-build
+import * as gamelift from '@aws-cdk-lib/aws-gamelift';
+
+const fleet = new gamelift.Fleet(this, 'Game server fleet', {
+  build: build,
+  runtimeConfiguration: {
+    gameSessionActivationTimeoutSeconds: 123,
+    maxConcurrentGameSessionActivations: 123,
+  }
+});
+fleet.runtimeConfiguration.addServerProcess('/local/game/GameLiftExampleServer.x86_64', '-logFile /local/game/logs/myserver1935.log -port 1935');
```

Comment:
Can we make this API more structured? For example, by creating a `ServerProcess` interface:

```ts
interface ServerProcess {
  readonly executablePath: string,
  readonly logFile: string,
  readonly port: number,
}
```